inspire/modules/deposit/static/js/deposit/stats/deposition_charts.js

```diff
@@ -0,0 +1,241 @@
+        type: this.type,
+        name: "Depositions",
+        data: this.data
+      }]
+    };
+
+    this.column_options = {
+      chart: {
+          type: 'column'
+      },
+      title: {
+          text: this.title
+      },
+      subtitle: {
+          text: this.subtitle
+      },
+      xAxis: {
+          categories: this.categories
+      },
+      yAxis: {
+          min: 0,
+          title: {
+              text: 'Times filled by users'
+          }
+      },
+      plotOptions: {
+          column: {
+              pointPadding: 0.2,
+              borderWidth: 0
+          }
+      },
+      series: massageColumnData(this.data)
+    };
+  }
+
+  DepositionChart.prototype = {
+    init: function(data){
+      var chart_options = {
+        title: "Submitted depositions",
+        type: "column",
+        data: data.columns_deposition_data,
+        categories: data.metadata_categories
+      };
+
+      $deposition_charts.highcharts(
+        new DepositionChart(chart_options).generate()
+      );
+    },
+
+    update: function(data){
+      var chart_type = $chart_type.val();
+      var chart_data = ((chart_type === "column") ?
+        data.columns_deposition_data : data.overall_depositions.pie);
+
+      var chart_options = {
+        title: "Submitted depositions",
+        type: chart_type,
+        data: chart_data
+      };
+
+      if(chart_type === "column") {
+        chart_options.categories = data.metadata_categories;
+      }
+
+      $deposition_charts.empty();
+      $deposition_charts.highcharts(
+        new DepositionChart(chart_options).generate()
+      );
+    },
+
+    filter: function(url){
+      var since_date = $('#since_date').val();
+      var until_date = $('#until_date').val();
+      var params = {};
+
+      if(since_date !== "")
+        params.since_date = since_date;
+      if(until_date !== "")
+        params.until_date = until_date;
+
+      if(!$.isEmptyObject(params)){
+        $.ajax({
+          url: url,
+          data: params
+        })
+          .done(function(data) {
+                var charts_data = {
+                  deposition_data: data.stats,
+                  overall_depositions: data.overall_metadata,
+                  columns_deposition_data: data.metadata_for_column,
+                  metadata_categories: data.metadata_categories,
+                }
+            $deposition_charts.empty();
+            new DepositionChart({}).update(charts_data);
+            if(Object.keys(charts_data.deposition_data).length > 1){
+              $('div[id^="chart-"]').parents('.row').show();
+              new DepositionChart({}).generate_individual(charts_data);
+            } else {
+              $('div[id^="chart-"]').parents('.row').hide();
+            }
+          });
+      }
+    },
+
+    reset_filter: function(url){
+      $('#since_date').val('');
+      $('#until_date').val('');
+
+      $.ajax({
+          url: url
+        })
+          .done(function(data) {
```

Comment:
It would be good to avoid duplicating the same code as above and instead group it in a function if possible